### PR TITLE
Patch release 2.3.4

### PR DESCRIPTION
**Pull request by jfy133**
Patch release for 2.3.4.

Mostly documentation update/fixes, and a couple of bug fixes.

Only code changes are moving gzipping back into pipes for adapterremoval, and insertion of a single parameter into the bowtie2 PE command.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
  - [x] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/eager/tree/master/.github/CONTRIBUTING.md)
  - [x] If necessary, also make a PR on the nf-core/eager _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by maxulysse — APPROVED** [review]
LGTM

**Review by apeltzer — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,28 @@
...
+- Updated DamageProfiler citation from bioRxiv to publication
+
+### `Dependencies`
+
+- Removed pinning of `tbb` (upstream bug in bioconda fixed)
+- Bumped `pigz` to 2.6 to fix rare stall bug when compressing data after AdapterRemoval
+- Bumped Bowtie2 to 2.4.2 to fix issues with `tbb` version
+
+### `Deprecated`
+
```
Comment (maxulysse) [inline CHANGELOG.md:26]:
```suggestion
```

I tend to remove empty sections, but not sure if it's how it really should be done

Comment (jfy133) [reply CHANGELOG.md:26]: I personally always carry all sections over for consistency, so will keep here. Dev choice I guess :shrug:

**Comment by jfy133** [comment]
Ok close enough, here we go!

**Pull request merged**